src/collectors/github_rate_limit.rs

```diff
@@ -135,34 +140,47 @@ impl GitHubRateLimit {
         let u = client
             .execute(req)
             .await
-            .unwrap()
+            .map_err(Error::from)?
```

Comment:
Is the `map_err` actually necessary?

Reply:
Well it is if we would like to uniform to a standard Error type (anyhow::Error).

Having every publlic method returning a potentially different Error struct with a different api makes my brain spin, i think wont hurt to adhere to an Error type at this "higher"-level, and if we need the inner type, at the call site we can always downcast to it (i believe). 

but i don't know if you see any technical reason to not doing it or if there's more to it. If so can you please elaborate?

Reply:
So, we indeed want to convert everything to `anyhow::Error`!

I'm saying that the `map_err` is not necessary because the `?` operator is doing that already behind the scenes. The compiler "transforms" `foo()?` roughly into this snippet of code:

```rust
match foo() {
    Ok(value) => value,
    Err(error) => return Err(error.into()),
}
```

...and thanks to the `.into()` the custrom error is converted into `anyhow::Error` automatically. If you try replacing `.await.map_err(Error::from)?` into `.await?` without changing the return type everything should still compile :)

Reply:
@pietroalbini you're completly right about that, I didn't knew about that. 

I've made the suggested changes. 👍 
thanks for the teaching!